Plugins/ConnectPluginUtilities/Generator.swift

```diff
@@ -139,7 +140,13 @@ extension Generator: SwiftProtobufPluginLibrary.CodeGenerator {
     }
 
     public var supportedEditionRange: ClosedRange<SwiftProtobuf.Google_Protobuf_Edition> {
-        return SwiftProtobufPluginLibrary.DescriptorSet.bundledEditionsSupport
+        let minEdition = max(
+            DescriptorSet.bundledEditionsSupport.lowerBound, Google_Protobuf_Edition.legacy
+        )
+        let maxEdition = min(
+            DescriptorSet.bundledEditionsSupport.upperBound, Google_Protobuf_Edition.edition2024
```

Comment:
It is sadly confusing in `descriptor.proto`, which shows 2024 as the latest edition. But 2024 isn't actually fully-defined yet. It's an experimental edition that is still in progress.

If you try to compile a file with it (using buf or the latest v27.2 of protoc) you'll get the following:
```
$ echo 'edition = "2024";' > test.proto && protoc test.proto -o /dev/null
test.proto:1:1: Edition 2024 is later than the maximum supported edition 2023
```

(You can try out 2024 in protoc by passing the `--experimental_editions` flag.)

Since it's incomplete (and 2023 changed quite a bit between early experimental releases like v24 and the final version in v27), we probably should hold off and use 2023 here.

Reply:
Ah, got it. This was also failing without the `min` here for the same reason, but that makes sense to not support 2024 until it's done. Updated